Android application terminates with exception

### Issue Description

When  a bad color value is specified (ex. navBarTextColor: '#fff'), android application terminates with the following exception:
```
11-18 09:07:16.846 29624 29624 D AndroidRuntime: Shutting down VM
11-18 09:07:16.846 29624 29624 E AndroidRuntime: FATAL EXCEPTION: main
11-18 09:07:16.846 29624 29624 E AndroidRuntime: Process: xxxxx, PID: 29624
11-18 09:07:16.846 29624 29624 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{xxxxx/com.reactnativenavigation.controllers.NavigationActivity}: java.lang.IllegalArgumentException: Unknown color
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.ActivityThread.-wrap11(ActivityThread.java)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:148)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: Caused by: java.lang.IllegalArgumentException: Unknown color
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.graphics.Color.parseColor(Color.java:226)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.reactnativenavigation.params.StyleParams$Color.parse(StyleParams.java:34)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.reactnativenavigation.params.parsers.StyleParamsParser.getColor(StyleParamsParser.java:190)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.reactnativenavigation.params.parsers.StyleParamsParser.parse(StyleParamsParser.java:31)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.reactnativenavigation.params.parsers.ScreenParamsParser.parse(ScreenParamsParser.java:30)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.reactnativenavigation.params.parsers.ActivityParamsParser.parse(ActivityParamsParser.java:16)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.reactnativenavigation.controllers.NavigationCommandsHandler.parseActivityParams(NavigationCommandsHandler.java:24)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at com.reactnativenavigation.controllers.NavigationActivity.onCreate(NavigationActivity.java:54)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:6237)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
11-18 09:07:16.846 29624 29624 E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
```
Application keeps terminating this way even after fixing the bad color value in JS source. Problem resolves after recompiling and reinstalling application with **react-native run-android**

### Steps to Reproduce / Code Snippets / Screenshots

```
const nav= {
  screen: {
    screen: 'Welcome',
    title: 'Welcome',
    navigatorStyle: {
      navBarTextColor: '#fff',
      statusBarTextColorScheme: 'light'
    }
  }
}
Navigation.startSingleScreenApp(nav)
```
### Environment
* React Native Navigation version: 2.0.0-experimental.141@18/11/2016
* React Native version: 0.37.0
* Platform(s): Android
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator, Android 6.0, Debug

--- a/android/app/src/main/java/com/reactnativenavigation/bridge/BundleConverter.java
+++ b/android/app/src/main/java/com/reactnativenavigation/bridge/BundleConverter.java
@@ -19,7 +19,7 @@ public class BundleConverter {
                     bundle.putBoolean(key, map.getBoolean(key));
                     break;
                 case Number:
-                    bundle.putDouble(key, map.getDouble(key));
+                    putNumber(bundle, map, key);
                     break;
                 case String:
                     bundle.putString(key, map.getString(key));
@@ -35,6 +35,14 @@ public class BundleConverter {
             }
         }
         return bundle;
+    }
+
+    private static void putNumber(Bundle bundle, ReadableMap map, String key) {
+        try {
+            bundle.putInt(key, map.getInt(key));
+        } catch (Exception e) {
+            bundle.putDouble(key, map.getDouble(key));
+        }
     }
 
     public static Bundle toBundle(ReadableArray array) {

--- a/android/app/src/main/java/com/reactnativenavigation/params/StyleParams.java
+++ b/android/app/src/main/java/com/reactnativenavigation/params/StyleParams.java
@@ -1,5 +1,6 @@
 package com.reactnativenavigation.params;
 
+import android.os.Bundle;
 import android.support.annotation.ColorInt;
 
 public class StyleParams {
@@ -27,11 +28,8 @@ public class StyleParams {
             return color;
         }
 
-        public static Color parse(String str) {
-            if (str == null) {
-                return new Color();
-            }
-            return new Color(android.graphics.Color.parseColor(str));
+        public static Color parse(Bundle bundle, String key) {
+            return bundle.containsKey(key) ? new Color(bundle.getInt(key)) : new Color();
         }
     }
 

--- a/android/app/src/main/java/com/reactnativenavigation/params/parsers/ContextualMenuButtonParamsParser.java
+++ b/android/app/src/main/java/com/reactnativenavigation/params/parsers/ContextualMenuButtonParamsParser.java
@@ -22,7 +22,7 @@ public class ContextualMenuButtonParamsParser extends TitleBarButtonParamsParser
         ContextualMenuButtonParams result = new ContextualMenuButtonParams();
         result.icon = ImageLoader.loadImage(button.getString("icon"));
         result.showAsAction = parseShowAsAction(button.getString("showAsAction"));
-        result.color = StyleParams.Color.parse(button.getString("color"));
+        result.color = StyleParams.Color.parse(button, "color");
         result.label = button.getString("label");
         result.index = (int) button.getDouble("index");
         return result;

--- a/android/app/src/main/java/com/reactnativenavigation/params/parsers/FabActionParamsParser.java
+++ b/android/app/src/main/java/com/reactnativenavigation/params/parsers/FabActionParamsParser.java
@@ -12,7 +12,7 @@ public class FabActionParamsParser extends Parser {
         fabActionParams.id = params.getString("id");
         fabActionParams.navigatorEventId = navigatorEventId;
         fabActionParams.icon = ImageLoader.loadImage(params.getString("icon"));
-        fabActionParams.backgroundColor = StyleParams.Color.parse(params.getString("backgroundColor"));
+        fabActionParams.backgroundColor = StyleParams.Color.parse(params, "backgroundColor");
         return fabActionParams;
     }
 }

--- a/android/app/src/main/java/com/reactnativenavigation/params/parsers/Parser.java
+++ b/android/app/src/main/java/com/reactnativenavigation/params/parsers/Parser.java
@@ -38,7 +38,7 @@ public class Parser {
     }
 
     protected StyleParams.Color getColor(Bundle bundle, String key, StyleParams.Color defaultColor) {
-        StyleParams.Color color = StyleParams.Color.parse(bundle.getString(key));
+        StyleParams.Color color = StyleParams.Color.parse(bundle, key);
         return color.hasColor() || defaultColor == null ? color : defaultColor;
     }
 }

--- a/android/app/src/main/java/com/reactnativenavigation/params/parsers/StyleParamsParser.java
+++ b/android/app/src/main/java/com/reactnativenavigation/params/parsers/StyleParamsParser.java
@@ -193,7 +193,7 @@ public class StyleParamsParser {
     }
 
     private StyleParams.Color getColor(String key, StyleParams.Color defaultColor) {
-        StyleParams.Color color = StyleParams.Color.parse(params.getString(key));
+        StyleParams.Color color = StyleParams.Color.parse(params, key);
         if (color.hasColor()) {
             return color;
         } else {

--- a/android/app/src/main/java/com/reactnativenavigation/views/ContentView.java
+++ b/android/app/src/main/java/com/reactnativenavigation/views/ContentView.java
@@ -1,7 +1,6 @@
 package com.reactnativenavigation.views;
 
 import android.content.Context;
-import android.util.Log;
 import android.view.View;
 
 import com.facebook.react.ReactRootView;
@@ -52,7 +51,6 @@ public class ContentView extends ReactRootView {
     protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
         super.onMeasure(widthMeasureSpec, heightMeasureSpec);
         int measuredHeight = viewMeasurer.getMeasuredHeight(heightMeasureSpec);
-        Log.i("Guy", "measuredHeight: " + measuredHeight + " " + getClass().getSimpleName());
         setMeasuredDimension(viewMeasurer.getMeasuredWidth(widthMeasureSpec),
                 measuredHeight);
     }

--- a/src/deprecated/platformSpecificDeprecated.android.js
+++ b/src/deprecated/platformSpecificDeprecated.android.js
@@ -116,45 +116,45 @@ function convertStyleParams(originalStyleObject) {
   }
 
   let ret = {
-    statusBarColor: originalStyleObject.statusBarColor,
-    topBarColor: originalStyleObject.navBarBackgroundColor,
+    statusBarColor: processColor(originalStyleObject.statusBarColor),
+    topBarColor: processColor(originalStyleObject.navBarBackgroundColor),
     topBarTransparent: originalStyleObject.navBarTransparent,
     topBarTranslucent: originalStyleObject.navBarTranslucent,
     collapsingToolBarImage: originalStyleObject.collapsingToolBarImage,
-    collapsingToolBarCollapsedColor: originalStyleObject.collapsingToolBarCollapsedColor,
+    collapsingToolBarCollapsedColor: processColor(originalStyleObject.collapsingToolBarCollapsedColor),
     titleBarHidden: originalStyleObject.navBarHidden,
     titleBarHideOnScroll: originalStyleObject.navBarHideOnScroll,
-    titleBarTitleColor: originalStyleObject.navBarTextColor,
-    titleBarSubtitleColor: originalStyleObject.navBarTextSubtitleColor,
-    titleBarButtonColor: originalStyleObject.navBarButtonColor,
-    titleBarDisabledButtonColor: originalStyleObject.titleBarDisabledButtonColor,
+    titleBarTitleColor: processColor(originalStyleObject.navBarTextColor),
+    titleBarSubtitleColor: processColor(originalStyleObject.navBarTextSubtitleColor),
+    titleBarButtonColor: processColor(originalStyleObject.navBarButtonColor),
+    titleBarDisabledButtonColor: processColor(originalStyleObject.titleBarDisabledButtonColor),
     backButtonHidden: originalStyleObject.backButtonHidden,
     topTabsHidden: originalStyleObject.topTabsHidden,
-    contextualMenuStatusBarColor: originalStyleObject.contextualMenuStatusBarColor,
-    contextualMenuBackgroundColor: originalStyleObject.contextualMenuBackgroundColor,
-    contextualMenuButtonsColor: originalStyleObject.contextualMenuButtonsColor,
+    contextualMenuStatusBarColor: processColor(originalStyleObject.contextualMenuStatusBarColor),
+    contextualMenuBackgroundColor: processColor(originalStyleObject.contextualMenuBackgroundColor),
+    contextualMenuButtonsColor: processColor(originalStyleObject.contextualMenuButtonsColor),
 
     drawBelowTopBar: !originalStyleObject.drawUnderNavBar,
 
-    topTabTextColor: originalStyleObject.topTabTextColor,
-    selectedTopTabTextColor: originalStyleObject.selectedTopTabTextColor,
+    topTabTextColor: processColor(originalStyleObject.topTabTextColor),
+    selectedTopTabTextColor: processColor(originalStyleObject.selectedTopTabTextColor),
     selectedTopTabIndicatorHeight: originalStyleObject.selectedTopTabIndicatorHeight,
-    selectedTopTabIndicatorColor: originalStyleObject.selectedTopTabIndicatorColor,
+    selectedTopTabIndicatorColor: processColor(originalStyleObject.selectedTopTabIndicatorColor),
 
-    screenBackgroundColor: originalStyleObject.screenBackgroundColor,
+    screenBackgroundColor: processColor(originalStyleObject.screenBackgroundColor),
 
     drawScreenAboveBottomTabs: !originalStyleObject.drawUnderTabBar,
 
-    bottomTabsColor: originalStyleObject.tabBarBackgroundColor,
-    bottomTabsButtonColor: originalStyleObject.tabBarButtonColor,
-    bottomTabsSelectedButtonColor: originalStyleObject.tabBarSelectedButtonColor,
+    bottomTabsColor: processColor(originalStyleObject.tabBarBackgroundColor),
+    bottomTabsButtonColor: processColor(originalStyleObject.tabBarButtonColor),
+    bottomTabsSelectedButtonColor: processColor(originalStyleObject.tabBarSelectedButtonColor),
     bottomTabsHidden: originalStyleObject.tabBarHidden,
     bottomTabsHiddenOnScroll: originalStyleObject.bottomTabsHiddenOnScroll,
     forceTitlesDisplay: originalStyleObject.forceTitlesDisplay,
-    bottomTabBadgeTextColor: originalStyleObject.bottomTabBadgeTextColor,
-    bottomTabBadgeBackgroundColor: originalStyleObject.bottomTabBadgeBackgroundColor,
+    bottomTabBadgeTextColor: processColor(originalStyleObject.bottomTabBadgeTextColor),
+    bottomTabBadgeBackgroundColor: processColor(originalStyleObject.bottomTabBadgeBackgroundColor),
 
-    navigationBarColor: originalStyleObject.navigationBarColor
+    navigationBarColor: processColor(originalStyleObject.navigationBarColor)
   }
 
   if (originalStyleObject.collapsingToolBarImage) {

